e2e/cypress/integration/enterprise/ldap_group/group_mentions_spec.js

```diff
@@ -0,0 +1,278 @@
+        });
+
+        // # Give the group a custom name different from its DisplayName attribute
+        cy.get('#groupMention').find('input').clear().type(groupName);
+
+        // # Click save button
+        saveConfig();
+
+        // * Assert that the group mention works as expected since the group is enabled and sysadmin always has permission to mention
+        assertGroupMentionEnabled(groupName);
+
+        // # Login as sysadmin and navigate to board group page
+        navigateToGroup(groupID);
+
+        // # Click the allow reference button
+        cy.findByTestId('allowReferenceSwitch').then((el) => {
+            el.find('button').click();
+        });
+
+        // # Click save button
+        saveConfig();
+
+        // * Assert that the group mention does not do anything since the group is disabled even though sysadmin has permission to mention
+        assertGroupMentionDisabled(groupName);
+    });
+
+    it('MM-23937 - Can restrict users from mentioning a group through the use_group_mentions permission', () => {
+        const groupName = `board_test_case_${Date.now()}`;
+
+        // # Login as sysadmin
+        cy.apiLogin('sysadmin');
+
+        // # Set group as allow reference = true with name groupName
+        cy.apiPatchGroup(groupID, {allow_reference: true, name: groupName});
+
+        // # Navigate to system scheme page
+        cy.visit('/admin_console/user_management/permissions/system_scheme');
+
+        // # Click reset to defaults, confirm and save
+        cy.findByTestId('resetPermissionsToDefault').click();
+        cy.get('#confirmModalButton').click();
+        saveConfig();
+
+        // # Login as a normal user
+        cy.apiLogin('user-1');
+
+        // * Assert that the group mention works as expected since the group is enabled and user has permission to mention
+        assertGroupMentionEnabled(groupName);
+
+        // # Login as sysadmin and navigate to system scheme
+        cy.apiLogin('sysadmin');
+        cy.visit('/admin_console/user_management/permissions/system_scheme');
+
+        // # Disable group mentions for users if enabled and save
+        cy.findByTestId('all_users-posts-use_group_mentions-checkbox').then((btn) => {
+            if (btn.hasClass('checked')) {
+                btn.click();
+            }
+        });
+        saveConfig();
+
+        // # Login as a regular member
+        cy.apiLogin('user-1');
+
+        // * Assert that the group mention does not do anything since the user does not have the permission to mention the group
+        assertGroupMentionDisabled(groupName);
+
+        // # Login as sysadmin and navigate to system scheme page
+        cy.apiLogin('sysadmin');
+        cy.visit('/admin_console/user_management/permissions/system_scheme');
+
+        // # Click reset to defaults confirm and save
+        cy.findByTestId('resetPermissionsToDefault').click();
+        cy.get('#confirmModalButton').click();
+        saveConfig();
```

Comment:
Can you please add this in the after block (or) afterEach block instead of adding it at the end of the test. This is because the specified steps to reset the System permissions may get skipped if the test fails in between. However if its included in the after block, the steps will definitely be executed.

Reply:
done